ontopy/ontology.py

```diff
@@ -1629,32 +1631,150 @@ def get_wu_palmer_measure(self, cls1, cls2):
+            Iterable,
+        ],
+        entitytype: Optional[
+            Union[
+                str,
+                ThingClass,
+                ObjectPropertyClass,
+                DataPropertyClass,
+                AnnotationPropertyClass,
+            ]
+        ] = "class",
+    ) -> Union[
+        ThingClass,
+        ObjectPropertyClass,
+        DataPropertyClass,
+        AnnotationPropertyClass,
+    ]:
         """Create and return new entity
 
-        Makes a new entity in the ontology with given parent(s).
-        Return the new entity.
+        Args:
+            name: name of the entity
+            parent: parent(s) of the entity
+            entitytype: type of the entity,
+                default is 'class' (str) 'ThingClass' (owlready2 Python class).
+                Other options
+                are 'data_property', 'object_property',
+                'annotation_property' (strings) or the
+                Python classes ObjectPropertyClass,
+                DataPropertyClass and AnnotationProperty classes.
+
+        Returns:
+            the new entity.
+
+        Throws exception if name consists of more than one word, if type is not
+        one of the allowed types, or if parent is not of the correct type.
+        By default, the parent is Thing.
 
-        Throws exception if name consists of more than one word.
         """
         if " " in name:
             raise LabelDefinitionError(
                 f"Error in label name definition '{name}': "
                 f"Label consists of more than one word."
             )
         parents = tuple(parent) if isinstance(parent, Iterable) else (parent,)
+
+        if entitytype == "class":
+            parenttype = owlready2.ThingClass
+        elif entitytype == "data_property":
+            parenttype = owlready2.DataPropertyClass
+        elif entitytype == "object_property":
+            parenttype = owlready2.ObjectPropertyClass
+        elif entitytype == "annotation_property":
+            parenttype = owlready2.AnnotationPropertyClass
+        elif entitytype in [
+            ThingClass,
+            ObjectPropertyClass,
+            DataPropertyClass,
+            AnnotationPropertyClass,
+        ]:
+            parenttype = entitytype
+        else:
+            raise EntityClassDefinitionError(
+                f"Error in entity type definition: "
+                f"'{entitytype}' is not a valid entity type."
+            )
+
         for thing in parents:
-            if not isinstance(thing, owlready2.ThingClass):
-                raise ThingClassDefinitionError(
+            if not isinstance(thing, parenttype):
+                raise EntityClassDefinitionError(
                     f"Error in parent definition: "
-                    f"'{thing}' is not an owlready2.ThingClass."
+                    f"'{thing}' is not an {parenttype}."
                 )
 
         with self:
             entity = types.new_class(name, parents)
```

Comment:
I may be lacking context but, isn't this missing some kind of reference to `entitytype`?

Reply:
Parenttype is set above depending on entitytype. It is done like this so that you can either choose to give entitytype as a str (which then needs to be converted) or give it as a Python class directly.